javatests/google/registry/rde/XjcToDomainResourceConverterTest.java

```diff
@@ -0,0 +1,326 @@
+      "domain_fragment_secdns.xml");
+
+  //List of packages to initialize JAXBContext
+  private static final String JAXB_CONTEXT_PACKAGES = Joiner.on(":")
+      .join(Arrays.asList(
+          "google.registry.xjc.contact",
+          "google.registry.xjc.domain",
+          "google.registry.xjc.host",
+          "google.registry.xjc.mark",
+          "google.registry.xjc.rde",
+          "google.registry.xjc.rdecontact",
+          "google.registry.xjc.rdedomain",
+          "google.registry.xjc.rdeeppparams",
+          "google.registry.xjc.rdeheader",
+          "google.registry.xjc.rdeidn",
+          "google.registry.xjc.rdenndn",
+          "google.registry.xjc.rderegistrar",
+          "google.registry.xjc.smd"));
+
+  @Rule
+  public final AppEngineRule appEngine = AppEngineRule.builder()
+      .withDatastore()
+      .build();
+
+  @Rule
+  public final ExceptionRule thrown = new ExceptionRule();
+
+  private Unmarshaller unmarshaller;
+
+  @Before
+  public void before() throws Exception {
+    createTld("example");
+    unmarshaller = JAXBContext.newInstance(JAXB_CONTEXT_PACKAGES)
+        .createUnmarshaller();
+  }
+
+  @Test
+  public void testConvertDomainResource() throws Exception {
+    ContactResource jd1234 = persistActiveContact("jd1234");
+    ContactResource sh8013 = persistActiveContact("sh8013");
+    ImmutableSet<DesignatedContact> expectedContacts = ImmutableSet.of(
+      DesignatedContact.create(DesignatedContact.Type.ADMIN, Key.create(sh8013)),
+      DesignatedContact.create(DesignatedContact.Type.TECH, Key.create(sh8013))
+    );
+
+    XjcRdeDomain xjcDomain = getDomain();
+    DomainResource domain = XjcToDomainResourceConverter.convertDomain(xjcDomain);
+    assertThat(domain.getFullyQualifiedDomainName()).isEqualTo("example1.example");
+    assertThat(domain.getRepoId()).isEqualTo("Dexample1-TEST");
+    // A DomainResource has status INACTIVE if there are no nameservers.
+    assertThat(domain.getStatusValues()).isEqualTo(ImmutableSet.of(StatusValue.INACTIVE));
+    assertThat(domain.getRegistrant().getName()).isEqualTo(jd1234.getRepoId());
+    assertThat(domain.getContacts()).isEqualTo(expectedContacts);
+    assertThat(domain.getCurrentSponsorClientId()).isEqualTo("RegistrarX");
+    assertThat(domain.getCreationClientId()).isEqualTo("RegistrarX");
+    assertThat(domain.getCreationTime()).isEqualTo(DateTime.parse("1999-04-03T22:00:00.0Z"));
+    assertThat(domain.getRegistrationExpirationTime())
+        .isEqualTo(DateTime.parse("2015-04-03T22:00:00.0Z"));
+    assertThat(domain.getGracePeriods().size()).isEqualTo(0);
+    assertThat(domain.getLastEppUpdateClientId()).isNull();
+    assertThat(domain.getLastEppUpdateTime()).isNull();
+  }
+
+  @Test
+  public void testConvertDomainResourceAddPeriod() throws Exception {
+    persistActiveContact("jd1234");
+    persistActiveContact("sh8013");
+    XjcRdeDomain xjcDomain = getDomain(DOMAIN_XML_ADD_PERIOD);
```

Comment:
Per above comment, this line would look like:

`XjcRdeDomain xjcDomain = getDomain(RdeTestData.get("domain_fragment_addPeriod.xml")); `

Even simpler, you could refactor getDomain() to take the filename directly, and rename it appropriately, so now it'd just look like:

`XjcRdeDomain xjcDomain = loadDomainFromRdeXml("domain_fragment_addPeriod.xml"); `